client/app/components/Row.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+class Row extends React.Component {
```

Comment:
Yes. However, the legend is really only acting as a container for these various rows, each of which update on their own, independent of the legend component. It seemed to make more sense to break them up into smaller pieces when I was adding in state to the rows.